WcaOnRails/app/webpacker/components/IncidentsLog.js

```diff
@@ -0,0 +1,307 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Dropdown,
+  Icon,
+  Input,
+  Table,
+} from 'semantic-ui-react';
+
+import { MiscTag, CompetitionTag, RegulationTag } from './Tags';
+import PaginationFooter from './PaginationFooter';
+import Loading from './Requests/Loading';
+import Errored from './Requests/Errored';
+
+import useLoadedData from '../lib/hooks/useLoadedData';
+import useDebounce from '../lib/hooks/useDebounce';
+import usePagination from '../lib/hooks/usePagination';
+
+import {
+  incidentsUrl,
+  newIncidentUrl,
+  incidentUrl,
+} from '../lib/requests/routes.js.erb';
+
+// incident helper functions //
+
+function incidentStatusClass({ resolved_at: resolvedAt }) {
+  return resolvedAt ? 'success' : 'warning';
+}
+
+function incidentStatusText({ resolved_at: resolvedAt }) {
+  return resolvedAt ? 'Resolved' : 'Pending';
+}
+
+function incidentDigestClass({ digest_worthy: digestWorthy, digest_sent_at: digestSentAt }) {
+  return digestWorthy && !digestSentAt ? 'warning' : 'success';
+}
+
+function incidentDigestText({ digest_worthy: digestWorthy, digest_sent_at: digestSentAt }) {
+  if (digestWorthy) {
+    return digestSentAt ? 'Sent' : 'Pending';
+  }
+  return '';
+}
+
+// constants //
+
+const DEBOUNCE_MS = 300;
+const SEARCH = 'search';
+const TAGS = 'tags';
+
+// incidents log //
+
+export default function IncidentsLog({
+  canManageIncidents = false,
+  canViewDelegateMatters = false,
```

Comment:
I did this, but it felt a bit awkward since there's no main root or app to wrap the provide around. And to avoid circular import/export, I had to define the context in a separate file.